5-AccessControl/1-call-api-roles/README.md

```diff
@@ -0,0 +1,487 @@
+For more information, see: [How to: Add app roles in your application and receive them in the token](https://docs.microsoft.com/azure/active-directory/develop/howto-add-app-roles-in-azure-ad-apps)
+
+#### Configure the service app (msal-node-api) to use your app registration
+
+Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `API\authConfig.json` file.
+1. Find the key `TenantId` and replace the existing value with your Azure AD tenant ID.
+1. Find the key `ClientId` and replace the existing value with the application ID (clientId) of **msal-node-api** app copied from the Azure portal.
+
+### Register the client app (msal-react-spa)
+
+1. Navigate to the [Azure portal](https://portal.azure.com) and select the **Azure AD** service.
+1. Select the **App Registrations** blade on the left, then select **New registration**.
+1. In the **Register an application page** that appears, enter your application's registration information:
+   - In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `msal-react-spa`.
+   - Under **Supported account types**, select **Accounts in this organizational directory only**.
+   - In the **Redirect URI (optional)** section, select **Single-page application** in the combo-box and enter the following redirect URI: `http://localhost:3000/`.
+1. Select **Register** to create the application.
+1. In the app's registration screen, find and note the **Application (client) ID**. You use this value in your app's configuration file(s) later in your code.
+1. Select **Save** to save your changes.
+1. In the app's registration screen, select the **API permissions** blade in the left to open the page where we add access to the APIs that your application needs.
+   - Select the **Add a permission** button and then,
+   - Ensure that the **My APIs** tab is selected.
+   - In the list of APIs, select the API `msal-node-api`.
+   - In the **Delegated permissions** section, select the **Access 'msal-node-api'** in the list. Use the search box if necessary.
+   - Select the **Add permissions** button at the bottom.
+
+#### Define Application Roles
+
+1. Still on the same app registration, select the **App roles** blade to the left.
+1. Select **Create app role**:
+    - For **Display name**, enter a suitable name, for instance **TaskAdmin**.
+    - For **Allowed member types**, choose **User**.
+    - For **Value**, enter **TaskAdmin**.
+    - For **Description**, enter **Admins can read any user's todo list**.
+1. Select **Create app role**:
+    - For **Display name**, enter a suitable name, for instance **TaskUser**.
+    - For **Allowed member types**, choose **User**.
+    - For **Value**, enter **TaskUser**.
+    - For **Description**, enter **Users can read and modify their todo lists**.
+1. Select **Apply** to save your changes.
```

Comment:
app roles on both client and api shoiuld not be needed. we assign users roles in api and when the client will get AT, the roles assigned to users in API will be present.

Reply:
🤦 ah, that should actually work. I'll give it a try